openapi-validator/src/main/java/org/ballerinalang/openapi/validator/ServiceValidator.java

```diff
@@ -257,47 +454,58 @@ private static void createListResourcePathSummary(List<ResourceValidationError>
     }
 
     /**
-     *  This for generate Dlog message with relevant type of errors.
+     *  This for generate Diagnostic message with relevant type of errors.
      * @param kind                  message type ned to display
-     * @param dLog                  diagnosticLog
      * @param resourcePathSummary   current validate ResourcePath Object
      * @param method                validate method
      * @param postErrors            list of validationErrors
      */
-    private static void generateDlogMessage(DiagnosticSeverity kind, DiagnosticLog dLog,
+    private static void generateDiagnosticMessage(DiagnosticSeverity kind,
                                             ResourcePathSummary resourcePathSummary,
                                             Map.Entry<String, ResourceMethod> method,
-                                            List<ValidationError> postErrors) {
+                                                  List<ValidationError> postErrors) {
 
         if (!postErrors.isEmpty()) {
             for (ValidationError postErr : postErrors) {
                 if (postErr instanceof TypeMismatch) {
-                    generateTypeMisMatchDlog(kind, dLog, resourcePathSummary, method, postErr);
+                    generateTypeMisMatchDiagnostic(kind, resourcePathSummary, method, postErr);
                 } else if (postErr instanceof MissingFieldInJsonSchema) {
-                    dLog.logDiagnostic(kind, method.getValue().getMethodPosition(),
+
+                    String[] error =
                             ErrorMessages.undocumentedFieldInRecordParam(postErr.getFieldName(),
                                     ((MissingFieldInJsonSchema) postErr).getRecordName(),
-                                    method.getKey(), resourcePathSummary.getPath()));
+                                    method.getKey(), resourcePathSummary.getPath());
+                    DiagnosticInfo diagnosticInfo = new DiagnosticInfo(error[0], error[1], kind);
+                    Diagnostic diagnostic = DiagnosticFactory.createDiagnostic(diagnosticInfo,
+                                    ((MissingFieldInJsonSchema) postErr).getLocation());
+                    validations.add(diagnostic);
+
                 } else if (postErr instanceof OneOfTypeValidation) {
                     if (!(((OneOfTypeValidation) postErr).getBlockErrors()).isEmpty()) {
-                        List<ValidationError> oneOferrorlist =
-                                ((OneOfTypeValidation) postErr).getBlockErrors();
+                        List<ValidationError> oneOferrorlist = ((OneOfTypeValidation) postErr).getBlockErrors();
```

Comment:
```suggestion
                        List<ValidationError> oneOfErrorList = ((OneOfTypeValidation) postErr).getBlockErrors();
```
Shouldn't we follow java naming conventions?